controllers/articles-controllers.js

```diff
@@ -1,17 +1,48 @@
 const {
     selectArticleById,
+    selectArticles,
+    selectCommentsByArticle
     
   } = require('../models/articles-models.js');
   
   exports.getArticleById = (req, res, next) => {
-    console.log("fjdijsoijdifjdoijdi")
-    const id = req.params.article_id
-    console.log(id)
-    selectArticleById(id).then((article) => {
+    
+    const {article_id} = req.params
+    
+    
+    selectArticleById(article_id).then((article) => {
+      console.log(article, "Article")
       
+      if (article.length===0 && !isNaN(article_id)) {res.status(404).send({ msg:'article doesn\'t exist' });
+}
+
       res.status(200).send( {article });
-      
-    }) .catch((err) => {
+
+   } ) .catch((err) => {
       next(err);
-    });
-    }
\ No newline at end of file
+    });}
+    
+    exports.getArticles = (req, res, next) => {
+        
+        selectArticles().then((results) => {
+
+          
+          res.status(200).send( {results });
+          
+        }) .catch((err) => {
+          next(err);
+        });
+        }
+        exports.getCommentsByArticle = (req, res, next) => {
+        const id =req.params.article_id;
+        console.log(id)
+        
+          selectCommentsByArticle(id).then((results) => {
+            if (results.length===0) {res.status(404).send({ msg:'article doesn\'t exist' });
```

Comment:
Deciding if there's an error or not should be done in the model, not the controller so the error can be sent to the controller and then onto your error handling middleware. Look at previous examples of Promise.reject() in the model for hints on how to do this